lib/CL/devices/spirv_queries.cc

```diff
@@ -0,0 +1,289 @@
+/* spirv_queries.h - sets up SPIR-V queries for the device
+ *
+ * Copyright (c) 2025 pocl developers
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+#include <algorithm>
+#include <vector>
+
+#include "spirv.hh"
+#include "spirv_queries.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+void
+pocl_setup_spirv_queries (cl_device_id dev)
+{
+  if (dev->num_ils_with_version == 0)
+    return;
+
+  std::vector<const char*> spirvExtendedInstructionSets;
+  std::vector<const char*> spirvExtensions;
+  std::vector<cl_uint> spirvCapabilities;
+
+  spirvExtendedInstructionSets.push_back("OpenCL.std");
+
+  spirvCapabilities.push_back(static_cast<cl_uint>(spv::Capability::Addresses));
+  spirvCapabilities.push_back(static_cast<cl_uint>(spv::Capability::Float16Buffer));
+  spirvCapabilities.push_back(static_cast<cl_uint>(spv::Capability::Int16));
+  spirvCapabilities.push_back(static_cast<cl_uint>(spv::Capability::Int8));
+  spirvCapabilities.push_back(static_cast<cl_uint>(spv::Capability::Kernel));
+  spirvCapabilities.push_back(static_cast<cl_uint>(spv::Capability::Linkage));
+  spirvCapabilities.push_back(static_cast<cl_uint>(spv::Capability::Vector16));
+
+  spirvCapabilities.push_back(static_cast<cl_uint>(spv::Capability::Int64));
```

Comment:
I think this capability (int64) should be if'ed on `dev->has_64bit_long`

Reply:
Fixed.